### PR TITLE
WIP:  :bug: Update known_types to add x-kubernetes-int-or-string for intStr

### LINKED ISSUE #304: IntOrString is not structural in generated OpenAPIv3 CRD
I'm using kubebuilder v2.0.0-beta.0 with a api Go type that includes `v1.PodSpec`.
The generated CRD yaml can be installed by fails to be structural with condition errors like:

```
        Forbidden: must be empty to be structural, spec.validation.openAPIV3Schema.properties[spec].properties[custom].items.properties[podSpec].properties[containers].items.properties[lifecycle].properties[postStart].properties[httpGet].properties[port].type:
        Required value: must not be empty for specified object fields, spec.validation.openAPIV3Schema.properties[spec].properties[custom].items.properties[podSpec].properties[containers].items.properties[lifecycle].properties[postStart].properties[tcpSocket].properties[port].anyOf[0].type:
        Forbidden: must be empty to be structural, spec.validation.openAPIV3Schema.properties[spec].properties[custom].items.properties[podSpec].properties[containers].items.properties[lifecycle].properties[postStart].properties[tcpSocket].properties[port].anyOf[1].type:
        Forbidden: must be empty to be structural, spec.validation.openAPIV3Schema.properties[spec].properties[custom].items.properties[podSpec].properties[containers].items.properties[lifecycle].properties[postStart].properties[tcpSocket].properties[port].type:
        Required value: must not be empty for specified object fields, spec.validation.openAPIV3Schema.properties[spec].properties[custom].items.properties[podSpec].properties[containers].items.properties[lifecycle].properties[preStop].properties[httpGet].properties[port].anyOf[0].type:
        Forbidden: must be empty to be structural, spec.validation.openAPIV3Schema.properties[spec].properties[custom].items.properties[
```

I would expect the auto-generated OpenAPIV3 definitions to be structural.

### PR DESCRIPTION
**Pull request by cliveseldon**
:bug: 

Fixes #304 

Note:

 * Have not committed `go.sum`- should I?
 * Its unclear what version controller-tools requires of dependant tools. I have taken the first update to apiextensions-apiserver that includes the commit: https://github.com/kubernetes/apiextensions-apiserver/commit/7551f6323c3a36e7e3515881792b78aa7b6107a9

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -25,9 +21,8 @@ require (
 	gopkg.in/check.v1 v1.0.0-20180628173108-788fd7840127 // indirect
 	gopkg.in/inf.v0 v0.9.1 // indirect
 	gopkg.in/yaml.v2 v2.2.2 // indirect
-	k8s.io/api v0.0.0-20190409021203-6e4e0e4f393b
-	k8s.io/apiextensions-apiserver v0.0.0-20190409022649-727a075fdec8
-	k8s.io/apimachinery v0.0.0-20190404173353-6a84e37a896d
-	k8s.io/klog v0.2.0 // indirect
+	k8s.io/api v0.0.0-20190620085009-fd6d441bb4ee
+	k8s.io/apiextensions-apiserver v0.0.0-20190620085545-9a1be6d1ab4f
+	k8s.io/apimachinery v0.0.0-20190513181300-e403c24f3a75
```
[the remark this answers is not in the document]
Comment (cliveseldon) [reply go.mod]: Well is part of the initial changes for v1.15. If you go further there are also changes for the Webhooks which seem non-backwards compatible. So I'm not sure how this is supposed to be migrated:

https://github.com/kubernetes/api/commit/77a6ece06b11c80143ab777fda1a851afd63f287

This is why I need advice on how you conform this project to breaking changes such as above?

**Comment by mengqiy** [comment]
This PR need to be on top of https://github.com/kubernetes-sigs/controller-tools/pull/276 to have `XIntOrString` field.
/hold
Hold til #276 is merged.

**Comment by marun** [comment]
#276 has merged

**Comment by tamalsaha** [comment]
@cliveseldon , can you rebase your pr? This is blocker for generating structural schema using kubebuilder. Thanks!

**Comment by tamalsaha** [comment]
FYI, I have opened a new pr with additional fixes https://github.com/kubernetes-sigs/controller-tools/pull/360

**Comment by tamalsaha** [comment]
This issue has been fixed in https://github.com/kubernetes-sigs/controller-tools/pull/360 . This pr can be closed.

**Pull request closed without merging**